packages/edit-site/src/components/page-patterns/grid.js

```diff
@@ -4,36 +4,56 @@
 import {
 	__unstableComposite as Composite,
 	__unstableUseCompositeState as useCompositeState,
+	__experimentalText as Text,
 } from '@wordpress/components';
+import { useRef } from '@wordpress/element';
+import { __, sprintf } from '@wordpress/i18n';
 
 /**
  * Internal dependencies
  */
 import GridItem from './grid-item';
 
-export default function Grid( { categoryId, label, icon, items } ) {
-	const composite = useCompositeState( { orientation: 'vertical' } );
+const PAGE_SIZE = 100;
+
+export default function Grid( { categoryId, items, ...props } ) {
+	const composite = useCompositeState( { wrap: true } );
+	const gridRef = useRef();
 
 	if ( ! items?.length ) {
 		return null;
 	}
 
+	const list = items.slice( 0, PAGE_SIZE );
+	const restLength = items.length - PAGE_SIZE;
+
 	return (
-		<Composite
-			{ ...composite }
-			role="listbox"
-			className="edit-site-patterns__grid"
-			aria-label={ label }
-		>
-			{ items.map( ( item ) => (
-				<GridItem
-					key={ item.name }
-					icon={ icon }
-					item={ item }
-					categoryId={ categoryId }
-					composite={ composite }
-				/>
-			) ) }
-		</Composite>
+		<>
+			<Composite
+				{ ...composite }
+				role="listbox"
+				className="edit-site-patterns__grid"
+				{ ...props }
+				ref={ gridRef }
+			>
+				{ list.map( ( item ) => (
+					<GridItem
+						key={ item.name }
+						item={ item }
+						categoryId={ categoryId }
+						{ ...composite }
+					/>
+				) ) }
+			</Composite>
+			{ restLength > 0 && (
```

Comment:
Instead of doing infinite scrolling or a load more button, we decided to just show this text for now and suggest the users search for older patterns. This might get changed if we decide to change to server-side filtering in the future though.